packages/compass-settings/src/components/settings/featureflags.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import type { RootState } from '../../stores';
+import { changeFieldValue } from '../../stores/settings';
+import type { SettingsListProps } from './settings-list';
+import { SettingsList } from './settings-list';
+import { pick } from '../../utils/pick';
+import preferences, { usePreference } from 'compass-preferences-model';
+
+const devFeatureFlagFields = [
+  'enableTextAsPipeline',
+  'showDevFeatureFlags',
+  'debugUseCsfleSchemaMap',
+] as const;
+
+export const publicFeatureFlagFields = ['lgDarkmode'] as const;
+
+const featureFlagFields = [
+  ...publicFeatureFlagFields,
+  ...devFeatureFlagFields,
+] as const;
+type FeatureFlagFields = typeof featureFlagFields[number];
+type FeatureFlagSettingsProps = Omit<
+  SettingsListProps<FeatureFlagFields>,
+  'fields'
+>;
+
+(globalThis as any).setShowDevFeatureFlags = async (
+  showDevFeatureFlags = true
+) => {
+  await preferences.savePreferences({ showDevFeatureFlags });
+};
+
+export function useShouldShowDevFeatureFlags(): boolean {
+  return (
+    usePreference('showDevFeatureFlags', React) ??
+    process.env.NODE_ENV === 'development'
+  );
+}
+
+export function useShouldShowFeatureFlagsSettings(): boolean {
+  return useShouldShowDevFeatureFlags() || publicFeatureFlagFields.length > 0;
```

Comment:
I checked this `showDevFeatureFlags` setting box on my local version and saved and now I don't see this option anymore - is that intended? How do I get this setting back?
![image](https://user-images.githubusercontent.com/1791149/206115426-3102c398-9db7-4447-a509-eae717659bbe.png)

Reply:
You can always call `setShowDevFeatureFlags(true);` in the devtools command line, that’s probably the easiest way?